.github/workflows/ci.yml

```diff
@@ -1,70 +1,173 @@
+      - name: Load dependency cache
+        uses: actions/cache@v4
+        with:
+          path: |
+            **/node_modules
+          key: ${{ runner.os }}-modules-${{ hashFiles('yarn.lock') }}
+      - name: Install dependencies
+        run: yarn install --frozen-lockfile --ignore-engines
+      - name: Run linter
+        run: yarn run lint
+      - name: Run depcheck
+        run: yarn run depcheck
+      - name: Test clean script
+        run: yarn run clean
+
+  test:
+    name: Unit tests
     strategy:
+      fail-fast: false
       matrix:
-        os: [ubuntu-latest, macos-latest, windows-latest]
+        os:
+          - macos-latest
+          - ubuntu-latest
+          - windows-latest
         node-version:
           - 18.x
           - 20.x
           - 22.x
+    runs-on: ${{ matrix.os }}
     steps:
+      - name: Ensure line endings are consistent
+        run: git config --global core.autocrlf input
+      - name: Check out repository
+        uses: actions/checkout@v4
       - name: Use Node.js ${{ matrix.node-version }}
         uses: actions/setup-node@v4
         with:
           node-version: ${{ matrix.node-version }}
+      - name: Load dependency cache
+        uses: actions/cache@v4
+        with:
+          path: |
+            **/node_modules
+          key: ${{ runner.os }}-modules-${{ hashFiles('yarn.lock') }}
+      - name: Install dependencies and build
+        run: yarn install --frozen-lockfile --ignore-engines
+      - name: Run unit tests
+        run: yarn run test-ci:unit
+        env:
+          NODE_OPTIONS: --max_old_space_size=4096
+      - name: Submit coverage results
+        uses: coverallsapp/github-action@v2
+        with:
+          github-token: ${{ secrets.github_token }}
+          flag-name: run-${{ matrix.node-version }}
+          parallel: true
+
+  systemtest:
+    name: System tests
+    strategy:
+      fail-fast: false
+      matrix:
+        node-version:
+          - 18.x
+          - 20.x
+          - 22.x
+        os:
+          - ubuntu-latest
+    runs-on: ${{ matrix.os }}
+    steps:
       - name: Ensure line endings are consistent
         run: git config --global core.autocrlf input
       - name: Check out repository
         uses: actions/checkout@v4
-      - name: Load cache
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v4
+        with:
+          node-version: ${{ matrix.node-version }}
+      - name: Load dependency cache
         uses: actions/cache@v4
         with:
           path: |
             **/node_modules
-          key: ${{ runner.os }}-test-modules-v1-${{ hashFiles('**/yarn.lock') }}
-      - name: Install dependencies
+          key: ${{ runner.os }}-modules-${{ hashFiles('yarn.lock') }}
+      - name: Install dependencies and build
         run: yarn install --frozen-lockfile --ignore-engines
-      - name: Build project
-        run: yarn run build
-      - name: Build project -- esModuleInterop true
```

Comment:
The tsconfig file of link traversal actually has the flag in it already, so this extra build step is not needed here. That is why I removed it.

Reply:
Hmm, right, that's because of a dep on one of the inrupt libs.
Now, I would still keep it, because I want to remove that flag from the config when possible.
So we need to keep this in the CI, otherwise we may forget re-adding it.

Reply:
Okay, it is back now, but in the tooling test step. That one felt like the correct place to add it.